lib/std/fmt.zig

```diff
@@ -926,7 +932,7 @@ fn formatSizeImpl(comptime base: comptime_int) type {
             const s = switch (magnitude) {
                 0 => buf[0..formatIntBuf(&buf, value, 10, .lower, .{})],
                 else => formatFloat(&buf, new_value, .{ .mode = .decimal, .precision = options.precision }) catch |err| switch (err) {
-                    error.BufferTooSmall => unreachable,
+                    error.BufferTooSmall => @panic("Buffer is too small to format float."),
```

Comment:
I'm not sure `@panic` is the right fix here (there's no precedence for using `@panic` in `std.fmt`). I think it would be better to keep this as unreachable and either: (1) fix the buffer size to ensure that it actually is unreachable, or (2) file an issue with a reproduction that hits the unreachable and address that in a follow-up PR.

Reply:
Oh alrighty! I'll add the test cases, and I can re-add the unreachable for now and possibly make an issue addressing it afterwards.

Reply:
I'll also look into replacing `min_buffer_size` with `bufferSize`